pkg/sparrow/run.go

```diff
@@ -89,14 +94,20 @@ func (s *Sparrow) Run(ctx context.Context) error {
 	go func() {
 		err := s.api(ctx)
 		if err != nil {
-			panic(fmt.Sprintf("Failed to start api: %v", err))
+			log.Error("Error running api server", "error", err)
 		}
```

Comment:
Shouldn't this immediately return the error?

Reply:
Short answer, no, it will break everything. We must rework how errors are handled in sparrow.

Long answer:

Unfortunately, our handling of errors in sparrow is done in multiple ways. The `api` uses an error channel, the sparrow itself has none, the checks have their own way of shutting down and handling errors.

Currently, if an error happens during the API server's operation, it is handled by the `api(ctx)` function of the sparrow and returned (if it's not non-expected error or the context is done). 

If we return the error when running the `api(ctx)` function, no graceful shutdown can be done, as the Sparrow's run function would return instantly with the error.

To properly handle errors in the sparrow, I think we should introduce one global shutdown function, that the sparrow will execute, at least for the components it manages (`api` & `targetManager`). General error handling in the Sparrow's `run` function should also maybe be handled with a channel, and all of it's components should start in a similar way (the api starts in a separate goroutine because it returns an error, the other functions just run and handle errors on their own).

Reply:
Ok then we should create another issue for this as it will be part of a bigger refactoring. Can you create the corresponding issue?